common/src/main/java/com/wynntils/models/statuseffects/type/StatusEffect.java

```diff
@@ -28,8 +33,49 @@ public StatusEffect(StyledText name, StyledText modifier, StyledText displayedTi
                 name,
                 StyledText.fromString(" "),
                 displayedTime);
+
+        if( modifier == StyledText.EMPTY ){
+            this.modifierSuffix = StyledText.EMPTY;
+            this.modifierValue = Optional.empty();
+        } else {
+            this.modifierSuffix = readModifierSuffix(this.modifier);
+            this.modifierValue = readModifierValue(this.modifier, this.modifierSuffix);
+        }
+    }
+
+    private StyledText readModifierSuffix(StyledText modifier){
+        if( modifier == StyledText.EMPTY) return StyledText.EMPTY;
+
+        String modifierString = modifier.getString();
+        int dashIndex = modifierString.indexOf('/');
+        int percentIndex = modifierString.indexOf('%');
+        int prefixIndex = Math.max(dashIndex, percentIndex);
+        
+        if( prefixIndex == -1 ){
+            return StyledText.EMPTY;
+        }
+        return StyledText.fromString(ChatFormatting.GRAY + modifierString.substring(prefixIndex));
+    }
+
+    private Optional<Double> readModifierValue(StyledText modifier, StyledText modifierSuffix){
+        String modifierStr = modifier.getStringWithoutFormatting();
+        if( modifierStr.isEmpty() ){
+            return Optional.empty();
+        }
+        int start = Math.max(modifierStr.indexOf('+') + 1, 0);
+
+        int end = Math.max(modifierStr.indexOf('%'), modifierStr.indexOf('/'));
+        if( end == -1 ) end = modifierStr.length(); // no suffix present
+        
+        if( end == start || end == 0 ){
+            return Optional.empty();
+        }
+
+        double val = Double.parseDouble(modifierStr.substring(start, end));
+        return Optional.of(val);
     }
```

Comment:
Private methods should be after public ones

Reply:
Also it seems like this method does the same thing. Just return a pair, the value and suffix.